BaseUnit.cpp

```diff
@@ -19,6 +21,83 @@ std::string BaseUnit::showStats() const
 {
 	return Name + ": HP:" + std::to_string(HP) + " DMG: " + std::to_string(DMG);
 }
+BaseUnit BaseUnit::parseUnit(const std::string& file_name)
+{
+	std::fstream infile(file_name);
+
+	if (!infile.is_open())
+	{
+		throw(file_name);
+	}
+
+	std::string line;
+	std::string nm = "";
+	int hp = -1;
+	int dm = -1;
+
+	//loop to identify unit attributes
+	while (!infile.eof())
+	{
+		std::getline(infile, line);
+
+		size_t pos;
+
+		//find name
+		if (nm == "")
+		{
+			pos = line.find("name");
+
+			if (pos != std::string::npos)
+			{
+				pos = line.find(":");
+				nm = line.substr(pos + 3, line.size());
```

Comment:
A `:` és a `"` karakter között akármennyi és akármilyen whitespace lehet. Azzal ki tudok egyezni, hogy soronként csak 1 key-value pár lehet, amik között nem lehet sortörés, de lehessen akármennyi szóköz/tabulátor közöttük.

Reply:
@hegyhati Szerinted is rendben van ez a követelmény a parserre?

Reply:
Módosítottam a parsert. Két különböző megközelítést használtam. Egyet a `name` -hez, ez az idézőjelekkel találja meg a kívánt tartalmat. A másik a `hp `és `dmg` számára simán kiszedi a whitespace-eket a string elejéről. Így félig-meddig a vesszős dolgot is megoldottam, mert exception-t dobok, ha nincs a végén.